src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -436,6 +442,23 @@ protected virtual void OnSaving(XmlDocument obj)
             if (handler != null) handler(obj);
         }
 
+
+        /// <summary>
+        /// Event that is fired when the workspace is collecting top level assemblies referenced by nodes
```

Comment:
Is it true that only one workspace at a time will ever handle these events and only one extension will ever invoke them?

If those invariants needs to hold for this to make sense - then I think we should watch for those and log or throw if either of those are not true.

IE - check the innovationList and make sure it's only one etc. - At a minimum - add a comment.